examples/ecs-target/main.tf

```diff
@@ -0,0 +1,325 @@
+}
+
+data "aws_iam_policy_document" "cloudtrail_policy" {
+  statement {
+    sid     = "AWSCloudTrailAclCheck20150319"
+    effect  = "Allow"
+    actions = ["s3:GetBucketAcl"]
+    resources = [
+      module.cloudtrail_bucket.s3_bucket_arn,
+      "${module.cloudtrail_bucket.s3_bucket_arn}/*"
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+  }
+
+  statement {
+    sid     = "AWSCloudTrailWrite20150319"
+    effect  = "Allow"
+    actions = ["s3:PutObject"]
+    resources = [
+      "${module.cloudtrail_bucket.s3_bucket_arn}/AWSLogs/${data.aws_caller_identity.current.account_id}/*"
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "s3:x-amz-acl"
+      values = [
+        "bucket-owner-full-control"
+      ]
+    }
+  }
+}
+
+module "source_bucket" {
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "~> 2.0"
+
+  bucket = "${random_pet.this.id}-source"
+}
+
+resource "aws_cloudtrail" "this" {
+  name                          = random_pet.this.id
+  s3_bucket_name                = module.cloudtrail_bucket.s3_bucket_id
+  include_global_service_events = true
+  is_multi_region_trail         = true
+
+  event_selector {
+    read_write_type           = "All"
+    include_management_events = true
+
+    data_resource {
+      type   = "AWS::S3::Object"
+      values = ["${module.source_bucket.s3_bucket_arn}/"]
+    }
+  }
+}
+
+data "aws_caller_identity" "current" {}
+
+######
+# ECS
+######
+
+module "ecs" {
+  source  = "terraform-aws-modules/ecs/aws"
+  version = "~> 3.0"
+
+  name = random_pet.this.id
+
+  capacity_providers = ["FARGATE", "FARGATE_SPOT", aws_ecs_capacity_provider.prov1.name]
+
+  default_capacity_provider_strategy = [{
+    capacity_provider = aws_ecs_capacity_provider.prov1.name # "FARGATE_SPOT"
+    weight            = "1"
+  }]
+}
+
+resource "aws_ecs_capacity_provider" "prov1" {
+  name = "prov1"
+
+  auto_scaling_group_provider {
+    auto_scaling_group_arn = module.asg.autoscaling_group_arn
+  }
+}
+
+resource "aws_ecs_service" "hello_world" {
+  name            = "hello_world-${random_pet.this.id}"
+  cluster         = module.ecs.ecs_cluster_id
+  task_definition = aws_ecs_task_definition.hello_world.arn
+  launch_type     = "EC2"
+
+  desired_count = 1
+
+  deployment_maximum_percent         = 100
+  deployment_minimum_healthy_percent = 0
+}
+
+resource "aws_ecs_task_definition" "hello_world" {
+  family = "hello_world-${random_pet.this.id}"
+
+  container_definitions = <<EOF
+[
+  {
+    "name": "hello_world-${random_pet.this.id}",
+    "image": "hello-world",
+    "cpu": 0,
+    "memory": 128,
+    "logConfiguration": {
+      "logDriver": "awslogs",
+      "options": {
+        "awslogs-region": "eu-west-1",
+        "awslogs-group": "hello_world-${random_pet.this.id}",
```

Comment:
Make tast_definition smaller by removing logging and cloudwatch_log_group resources.